base/lnls-prune-artifacts.sh

```diff
@@ -0,0 +1,122 @@
+#!/usr/bin/env bash
+
+set -eu
+
+target_paths=$@
+
+get_linked_libraries() {
+    targets=$@
+
+    executables=$(find $targets -type f -executable -print)
+
+    libs="$(ldd $executables 2>/dev/null | grep '=>' | cut -d'=' -f 1)"
+    for lib in $libs; do
+        echo ${lib%.so*}.so
+    done | sort | uniq
+}
+
+find_shared_libs() {
+    search_paths=$@
+
+    libs=$(find $search_paths -type f -executable -name *.so* -print)
+
+    echo "$libs" | grep -E "*.so(.[0-9]+)*$" | sort | uniq
+}
+
+get_used_epics_modules() {
+    targets=$@
+
+    executables=$(find $targets -type f -executable -print)
+    libs="$(ldd $executables 2>/dev/null)"
+
+    modules=()
+    epics_libs=$(echo "$libs" | grep -oE "$EPICS_MODULES_PATH/.*/*.so")
+    for lib in $epics_libs; do
+        module=${lib%/lib/*/*lib*so}
+        echo $module
+    done | sort | uniq
+}
+
+get_all_epics_modules() {
+    release_defs=$(cat ${EPICS_RELEASE_FILE} | grep = | cut -d'=' -f 2)
+
+    modules="$(echo "$release_defs" | grep $EPICS_MODULES_PATH)"
+
+    for module in $modules; do
+        echo $module
+    done
+}
+
+remove_static_libs() {
+    for target; do
+        libs=$(find $target -type f -name *.a -print)
+
+        if [ -n "$libs" ]; then
+            size=$(du -hsc $libs | tail -n 1 | cut -f 1)
+
+            echo "Removing static libraries from $target ($size)"
```

Comment:
I'm collapsing all static libraries. Should we list the removed files instead?

I decided to let it collapsed because it is simpler to implement and no application should need them after build.

Reply:
I'm okay with them being collapsed.